### PR TITLE
Only display warning banner in batch changes if no associated legacy OR new webhook found.

### PR DESCRIPTION
**Pull request by mollylogue**
When determining whether or not to display a banner in the batch changes UI to configure webhooks, we currently run a query against the `has_webhooks` column in the `external_services` table. This PR updates this to also check if there is an associated row in the `webhooks` table. So, we will only display the banner if there is no webhook set up (either in the legacy manner via the external service config, or in the webhooks table). We match a webhook to an external service repo via the url (i.e. the `code_host_urn` in the `webhooks` table and the `external_service_id` in the `repo` table.)

Closes https://github.com/sourcegraph/sourcegraph/issues/44732

## Test plan
Validate via the UI that this works as expected

### CODE REVIEW

File: enterprise/internal/batches/store/codehost_test.go [hunk 1]
```diff
@@ -105,26 +107,48 @@ func testStoreCodeHost(t *testing.T, ctx context.Context, s *Store, clock bt.Clo
...
+					},
+					{
+						ExternalServiceType: extsvc.TypeGitLab,
+						ExternalServiceID:   "https://gitlab.com/",
+						HasWebhooks:         false,
+					},
+				}
+				if diff := cmp.Diff(have, want); diff != "" {
+					t.Fatalf("Invalid code hosts returned. %s", diff)
+				}
```
Comment (ryanslade) [inline enterprise/internal/batches/store/codehost_test.go:130]:
```suggestion
                 assert.Equal(t, want, have)
```

File: enterprise/internal/batches/store/codehost_test.go [hunk 2]
```diff
@@ -105,26 +107,48 @@ func testStoreCodeHost(t *testing.T, ctx context.Context, s *Store, clock bt.Clo
...
+				want := []*btypes.CodeHost{
+					{
+						ExternalServiceType: extsvc.TypeGitLab,
+						ExternalServiceID:   "https://gitlab.com/",
+						HasWebhooks:         false,
+					},
+				}
+				if diff := cmp.Diff(have, want); diff != "" {
+					t.Fatalf("Invalid code hosts returned. %s", diff)
+				}
```
Comment (ryanslade) [inline enterprise/internal/batches/store/codehost_test.go:150]:
```suggestion
				assert.Equal(t, want, have)
```

**Review by ryanslade — APPROVED** [review]

File: enterprise/internal/batches/store/codehost.go [hunk 3]
```diff
@@ -127,7 +127,7 @@ func listCodeHostsQuery(opts ListCodeHostsOpts) *sqlf.Query {
 
 	var aggregatePreds []*sqlf.Query
 	if opts.OnlyWithoutWebhooks {
-		aggregatePreds = append(aggregatePreds, sqlf.Sprintf("has_webhooks_count = 0"))
+		aggregatePreds = append(aggregatePreds, sqlf.Sprintf("has_webhooks_count = 0 AND external_service_id NOT IN (SELECT code_host_urn FROM webhooks)"))
```
Comment (mrnugget) [inline enterprise/internal/batches/store/codehost.go:130]: We don't need to query all of them :)

```suggestion
		aggregatePreds = append(aggregatePreds, sqlf.Sprintf("has_webhooks_count = 0 AND external_service_id NOT IN (SELECT DISTINCT(code_host_urn) FROM webhooks)"))
```

**Review by mrnugget — APPROVED** [review]
Approving but I think @sourcegraph/batchers should alsot ake a look. I also think the `DISTINCT` is something we should have.

**Review by eseliger — APPROVED** [review]

**Review by LawnGnome — APPROVED** [review]
Agreed with Thorsten; otherwise LGTM. :+1:

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Comment by mrnugget** [comment]
I just hopped on this branch and pushed a commit to address the feedback in the PR. We're doing webhook testing today, so I want all the webhook code to already be in `main` when we test.

**Comment by mollylogue** [comment]
@mrnugget thx for doing this. I meant to merge yesterday but got caught up with customer issues. 🙏

**Pull request merged**